crates/miden-lib/src/transaction/mod.rs

```diff
@@ -227,6 +231,17 @@ impl TransactionKernel {
             merkle_path.inner_nodes(account_id.prefix().as_u64(), account_header.commitment())?,
         );
 
+        // Load merkle nodes for storage maps
+        for proof in storage_proofs {
+            // Extend the merkle store and map with the storage maps
+            advice_inputs.extend_merkle_store(
+                proof.path().inner_nodes(proof.leaf().index().value(), proof.leaf().hash())?,
+            );
+            // Populate advice map with Sparse Merkle Tree leaf nodes
+            advice_inputs
+                .extend_map(core::iter::once((proof.leaf().hash(), proof.leaf().to_elements())));
+        }
```

Comment:
Was this a bug previously?

Reply:
Do you mean the fact that it was not there? I think it was more expected to be provided separately (we did so in the client). Storage map keys were not initially supported on the first iterations and I think this function was moved from the client to here and then never updated, it seems.